qiskit/dagcircuit/dagnode.py

```diff
@@ -251,13 +251,16 @@ class DAGOpNode(DAGNode):
 
     __slots__ = ["op", "qargs", "cargs", "sort_key"]
 
-    def __init__(self, op, qargs: Iterable[Qubit] = (), cargs: Iterable[Clbit] = ()):
+    def __init__(self, op, qargs: Iterable[Qubit] = (), cargs: Iterable[Clbit] = (), dag=None):
         """Create an Instruction node"""
         super().__init__()
         self.op = op
         self.qargs = tuple(qargs)
         self.cargs = tuple(cargs)
-        self.sort_key = str(self.qargs)
+        if dag is not None:
+            self.sort_key = str([dag.find_bit(q).index for q in self.qargs])
```

Comment:
Ah I missed that in your earlier comment yeah let's zero pad it. I'll add this after my asv run finishes.

Reply:
Also yeah clbits should probably be here too, I"ll add cargs into the iterator too

Reply:
I think we probably need a marker before the cargs start (probably we only add that to the string _if_ there are cargs, because most ops won't involve cargs and we can avoid the second string allocation), to avoid the case that an op on `q0, q1` would generate the same key as an op on `c0, c1`, and the topological ordering need not distinguish those two into layers.

Reply:
Yeah, let's not worry about it.  I suspect / hope that we'll change how `cargs` are handled before we actually get a problem here.